prow/pod-utils/decorate/podspec.go

```diff
@@ -593,6 +594,17 @@ func decorate(spec *coreapi.PodSpec, pj *prowapi.ProwJob, rawEnv map[string]stri
 	// TODO(fejta): we should pass around volume names rather than forcing particular mount paths.
 
 	rawEnv[artifactsEnv] = artifactsPath
+
+	if gcsConfig := pj.Spec.DecorationConfig.GCSConfiguration; gcsConfig != nil {
+		jobSpec := downwardapi.NewJobSpec(pj.Spec, rawEnv[downwardapi.BuildIDEnv], pj.Name)
+		_, gcsPath, _ := gcsupload.PathsForJob(gcsConfig, &jobSpec, "")
+		rawEnv[gcsArtifactsDirEnv] = "gs://" + path.Join(
+			gcsConfig.Bucket,
```

Comment:
According to: https://github.com/kubernetes/test-infra/blob/master/prow/apis/prowjobs/v1/types.go#L536
that could be:
	// * a GCS bucket: with gs:// prefix
	// * a S3 bucket: with s3:// prefix
	// * a GCS bucket: without a prefix (deprecated)
So prefixing that with gs:// will be wrong in two first cases.

Reply:
Could you add a test case for Bucket="gs://..." case?

Reply:
Changed the logic of the env var creation accordingly.
I added the test case with the bucket name starting with "gs://" as well.

Reply:
Could you handle s3:// case as well (in general, any link with the scheme specified?

Reply:
OK, I found s3 condition in url above. I am still worried about generality of that.
I would suggest to extract scheme in one place and handle cases (lack of, gs, other)

Reply:
Applied an appropriate change. Now, if no scheme is provided, it's set to `gs`, otherwise it's being left as it is.